block-util/src/config/mod.rs

```diff
@@ -66,15 +62,13 @@ pub trait BlockchainConfigExt {
 }
 
 impl BlockchainConfigExt for BlockchainConfig {
-    fn validate_params(
-        &self,
-        _relax_par0: bool,
-        _mandatory_params: Option<Dict<u32, ()>>,
-    ) -> Result<bool> {
-        // TODO: refer to https://github.com/everx-labs/ever-block/blob/master/src/config_params.rs#L452
-        // STUB: currently should not be invoked in prototype
-        // todo!()
-        Ok(true)
+    fn validate_params(&self) -> bool {
+        match self.get::<ConfigParam0>() {
+            Ok(Some(config_address)) if config_address == self.address => {
+                self.get_mandatory_params().is_ok() && self.get_critical_params().is_ok()
```

Comment:
In this case we should get the list of mandatory params and check that config contains this params and they can be parsed. We should merge lists of mandatory and critical params and check them all.

Reply:
Critical params are just some params that have a special requirements for update proposals. They can be absent